homeassistant/components/recorder/util.py

```diff
@@ -475,13 +514,27 @@ def setup_connection_for_dialect(
             # as its persistent and isn't free to call every time.
             result = query_on_connection(dbapi_connection, "SELECT sqlite_version()")
             version_string = result[0][0]
-            version = _extract_version_from_server_response(version_string)
+            version = _extract_version_from_server_response_or_raise(version_string)
 
-            if not version or version < MIN_VERSION_SQLITE:
+            if version < MIN_VERSION_SQLITE:
                 _fail_unsupported_version(
```

Comment:
We could consider renaming this to `_raise_if_version_unsupported` or something, just to be aligned with suffixing `_extract_version_from_server_response` with `_or_raise`

Reply:
777e02552ab3d91d6dd4671d085a28226a213c56